src/statistics.jl

```diff
@@ -857,7 +879,6 @@ function StatsBase.mean_and_var(
             return y, v
         end
     end
-    return y, v
```

Comment:
Is this function still type stable after removing this line? Compiler may think it could return `nothing` after removing this line.

Reply:
But that line is really never reached, so how can we resolve this? Look at the injectivty Radius for y upfront, if that does not trigger the return after that, one of the ifs _in_ the loop will trigger a return.

Reply:
I think the last line would be reached if `method.radius` in `NaN`. There may be other cases for custom floating point types. After the loop we need to either return `y, v` or throw an error.

Reply:
Ok, should we catch that upfront? Because we don't have a chance to test this, have we? So this line that you currently propose at the end will never be covered by a test then?

Reply:
We can't really catch that upfront, and we could test the last line by calling this method with geodesic interpolation within radius `NaN`, which is easy enough. I'm just not sure if this should result in an error or just return `y, v`.

Reply:
I am not sure which variant is best, I just wanted to get a line of test covered or remove it if it is never reached, I did not expect that it gets so difficult.

Reply:
So maybe let's just leave that line there for now and add this issue to #321 ?

Reply:
ok.